cylc/flow/task_pool.py

```diff
@@ -928,13 +946,17 @@ def warn_stop_orphans(self):
     def is_stalled(self):
         """Return True if the workflow is stalled.
 
-        A workflow is stalled if the active pool contains only unhandled
-        failed tasks.
+        A workflow is stalled if it is not held and the active pool contains
+        only unhandled failed tasks and un-released runahead tasks.
+
         """
+        self.release_runahead_tasks()
```

Comment:
Sorry, I don't follow?

This is here in case there are runahead-limited tasks that could be released (because they are now below the runahead limit) at the time of stall check, in which case, we're not stalled.  Unfortunately I don't recall why I thought that was necessary at this point. ...  I'll go back and see if I can see why.